presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketing.java

```diff
@@ -209,9 +213,45 @@ private static int hashBytes(int initialValue, Slice bytes)
         return Optional.of(new HiveBucketHandle(bucketColumns.build(), hiveBucketProperty.get().getBucketCount()));
     }
 
-    public static Optional<HiveBucket> getHiveBucket(Table table, Map<ColumnHandle, NullableValue> bindings)
+    public static List<HiveBucket> getHiveBucketNumbers(Table table, TupleDomain<ColumnHandle> effectivePredicate)
     {
-        if (!table.getStorage().getBucketProperty().isPresent() || bindings.isEmpty()) {
+        if (!table.getStorage().getBucketProperty().isPresent()) {
+            return ImmutableList.of();
+        }
+
+        Optional<Map<ColumnHandle, NullableValue>> maybeBindings = TupleDomain.extractFixedValues(effectivePredicate);
```

Comment:
Done. I presume this is naming convention in Presto - do not distinguish Optional somehow?